src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -174,6 +176,61 @@ _mongoc_write_command_init_insert (mongoc_write_command_t *command, /* IN */
 }
 
 
+// `_mongoc_write_command_init_insert_one_idl` returns the inserted ID in `inserted_id`.
+// Only called by mongoc_collection_insert_one.
+void
+_mongoc_write_command_init_insert_one_idl (mongoc_write_command_t *command,
+                                           const bson_t *document,
+                                           const bson_t *cmd_opts,
+                                           bson_t *insert_id,
+                                           int64_t operation_id)
+{
+   mongoc_bulk_write_flags_t flags = MONGOC_BULK_WRITE_FLAGS_INIT;
+
+   ENTRY;
+
+   BSON_ASSERT (command);
```

Comment:
```suggestion
   BSON_ASSERT_PARAM (command);
   BSON_ASSERT_PARAM (document);
   BSON_ASSERT_PARAM (cmd_opts);
   BSON_ASSERT_PARAM (insert_id);
```

Suggest using the newer `BSON_ASSERT_PARAM` macros on pointer arguments that are expected not to be NULL. If the caller incorrectly passes NULL, the macro prints an informative message. Example:

```
The parameter: cmd_opts, in function _mongoc_write_command_init_insert_one_idl, cannot be NULL
```